packages/mui-system/src/ThemeProvider/ThemeProvider.js

```diff
@@ -61,11 +62,13 @@ function ThemeProvider(props) {
 
   const engineTheme = useThemeScoping(themeId, upperTheme, localTheme);
   const privateTheme = useThemeScoping(themeId, upperPrivateTheme, localTheme, true);
+  const rtl = engineTheme.direction === 'rtl';
+  const RtlProvider = rtl === true ? SystemRtlProvider : React.Fragment;
```

Comment:
This is not needed after your latest change

Reply:
I wanted to avoid using the context if not necessary, but maybe it's not that big a deal. I'll update to simplify the code.